feat: update UI to show estimated deletion date instead of default

walbertus opened the pull request:
this is done by adding new job schedule to recalculate the estimate for deletion date. This function also called during the cleanup job

close #203

walbertus committed:
feat: add job to estimate media deletion date

walbertus committed:
feat: replace the deletion date in UI with estimated deletion date

walbertus committed:
chore: update naming related to deletion estimation

walbertus committed:
fix: disable estimate instant after start

walbertus commented:
@jon4hz please take a look at this when you're available, thanks

jon4hz commented on internal/api/models/converter.go:
```diff
@@ -9,14 +9,15 @@ import (
+		MediaType:         MediaType(m.MediaType),
+		LibraryName:       m.LibraryName,
+		FileSize:          m.FileSize,
+		DefaultDeleteAt:   m.DefaultDeleteAt,
```
It would be nice if `EstimatedDeleteAt` defaults to ` DefaultDeleteAt` in case the estimated delete time is zero. Otherwise the UI will show something like "Deletes 739603 days ago " until the scheduled job runs and calculates the estimated delete time.

jon4hz commented on internal/api/models/converter.go:
```diff
@@ -48,20 +49,21 @@ func ToUserMediaItems(items []database.Media, cfg *config.Config) []UserMediaIte
+		MediaType:         MediaType(m.MediaType),
+		RequestedBy:       m.RequestedBy,
+		DefaultDeleteAt:   m.DefaultDeleteAt,
+		EstimatedDeleteAt: m.EstimatedDeleteAt,
```
and here

jon4hz commented on internal/engine/scheduler.go:
```diff
@@ -49,6 +49,19 @@ func (e *Engine) setupJobs() error {
+		"0 */6 * * *", // Every 6 hours
+		estimateDeletionJobDef,
+		e.runEstimateDeletionsJob,
+		false,
```
Is there any downside to setting this to `true` so that the job executes on startup?

walbertus committed:
fix: default the deleted at to prevent zero

walbertus committed:
chore: rerun make build

walbertus committed:
fix: apply Copilot-suggested fixes

- Simplify redundant function in internal/database/media.go
- Checked internal/engine/estimate.go for hidden Unicode (none found)

walbertus committed:
fix: apply Copilot GORM zero-value suggestion for all Update calls

- Add Select() to Updates() calls to ensure zero values are saved
- Affects SetMediaEstimatedDeleteAt, SetMediaProtectedUntil, and MarkMediaAsUnkeepable
- Without Select(), GORM skips zero-value fields (zero time, nil pointers, false bools)
- Fixes potential data inconsistency when clearing or setting zero values

walbertus committed:
chore: rerun make build

walbertus commented:
I'll recheck this in a moment ya, it seem my last force push is wrong. It's been few months, I need time to revisit this

jon4hz commented:
Hi @walbertus,

Thanks for rebasing, and sorry that I couldn't give this PR more attention sooner. I'll try to test this in the next few days and if everything looks fine, we'll :shipit:

walbertus committed:
fix: restore disk usage guard, fix err overwrite, add godoc

Mavyre commented:
I'm interested in that feature, I initially thought I misconfigured something. When can we expect this to be merged?

hacshacdgacs commented:
Any chance of this getting attention again? Went down a rabbit-hole thinking my policies weren't working only to realise that it was just the UI not updating to reflect.
